packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -74,6 +82,16 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
         return this.id;
     }
 
+    protected get serializer(): IFluidSerializer {
+        assert(!this._isSummarizing, "SummarySerializer should be used for serializing data during summary.");
```

Comment:
I'd put a comment on the intention, as well as when it will be wrong assert.
For example, if we implement delay-loading in DDSs, then DDS may load and de-serialize content within shanshot() call, and thus hit it, even though code is correct. We do not do it today, so this assert is useful, but at some point it may not work